enginetest/queries/script_queries.go

```diff
@@ -2509,6 +2509,90 @@ var ScriptTests = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "year type behavior",
+		SetUpScript: []string{
+			"create table t (pk int primary key, col1 year);",
+		},
+		Assertions: []ScriptTestAssertion{
+			// 1901 - 2155 are interpreted as 1901 - 2155
+			{
+				Query:    "INSERT INTO t VALUES (1, '1901'), (2, 1901);",
+				Expected: []sql.Row{{sql.NewOkResult(2)}},
+			},
+			{
+				Query:    "INSERT INTO t VALUES (3, '2000'), (4, 2000);",
+				Expected: []sql.Row{{sql.NewOkResult(2)}},
+			},
+			{
+				Query:    "INSERT INTO t VALUES (5, '2155'), (6, 2155);",
+				Expected: []sql.Row{{sql.NewOkResult(2)}},
+			},
+			// 1 - 69 are interpreted as 2001 - 2069
+			{
+				Query:    "INSERT INTO t VALUES (7, '1'), (8, 1);",
+				Expected: []sql.Row{{sql.NewOkResult(2)}},
+			},
+			{
+				Query:    "INSERT INTO t VALUES (9, '35'), (10, 35);",
+				Expected: []sql.Row{{sql.NewOkResult(2)}},
+			},
+			{
+				Query:    "INSERT INTO t VALUES (11, '69'), (12, 69);",
+				Expected: []sql.Row{{sql.NewOkResult(2)}},
+			},
+			// 70 - 99 are interpreted as 1970 - 1999
+			{
+				Query:    "INSERT INTO t VALUES (13, '70'), (14, 70);",
+				Expected: []sql.Row{{sql.NewOkResult(2)}},
+			},
+			{
+				Query:    "INSERT INTO t VALUES (15, '85'), (16, 85);",
+				Expected: []sql.Row{{sql.NewOkResult(2)}},
+			},
+			{
+				Query:    "INSERT INTO t VALUES (17, '99'), (18, 99);",
+				Expected: []sql.Row{{sql.NewOkResult(2)}},
+			},
+			// '0', and '00' are interpreted as 2000
+			{
+				Query:    "INSERT INTO t VALUES (19, '0'), (20, '00');",
+				Expected: []sql.Row{{sql.NewOkResult(2)}},
+			},
+			// 0 is interpreted as 0000
+			{
+				Query:    "INSERT INTO t VALUES (21, 0)",
```

Comment:
what happens if you `INSERT INTO t VALUES (22, '0')`? Maybe not relevant for this specific PR, but seems like lots of potential edge cases of weird strings people could pass as years.

Reply:
It counts as the year 2000. See case 19, and 20.